src/clj/puppetlabs/services/jruby/jruby_puppet_internal.clj

```diff
@@ -95,44 +91,55 @@
   [ruby-load-path :- [schema/Str]]
   (cons ruby-code-dir ruby-load-path))
 
-(defn prep-scripting-container
-  [scripting-container ruby-load-path gem-home]
-  ; Note, this behavior should remain consistent with new-main
-  (doto scripting-container
+(schema/defn ^:always-validate get-compile-mode :- RubyInstanceConfig$CompileMode
+  [config-compile-mode :- jruby-schemas/SupportedJRubyCompileModes]
+  (case config-compile-mode
+    :jit RubyInstanceConfig$CompileMode/JIT
+    :force RubyInstanceConfig$CompileMode/FORCE
+    :off RubyInstanceConfig$CompileMode/OFF))
+
+(schema/defn ^:always-validate init-jruby-config :- jruby-schemas/ConfigurableJRuby
+  "Applies configuration to a JRuby... thing.  See comments in `ConfigurableJRuby`
+  schema for more details."
+  [jruby-config :- jruby-schemas/ConfigurableJRuby
```

Comment:
That seems reasonable but I'm not sure if I consider it urgent enough to gate this PR on.  As much as I hate this code I do think that it would explode pretty obviously if the upstream API changed, don't you?  I'm definitely willing to file a follow-up ticket that suggests what you said, and could be talked into gating this PR on that work, but would probably lean towards not blocking this one.